speculos/mcu/ocr.py

```diff
@@ -204,10 +226,27 @@ def analyze_bitmap(self, data: bytes):
         # Space is now encoded as an empty character (no 'space' wasted :)
         if len(bitmap) == 0:
             char = ' '
+            legacy = False
         else:
-            char = find_char_from_bitmap(bitmap)
+            legacy, char = find_char_from_bitmap(bitmap)
         if char:
-            if self.events and y <= (self.events[-1].y + self.events[-1].h):
+            if legacy:
+                # char is not from a loaded JSON font => keep legacy behaviour
+                if self.events and y <= self.events[-1].y:
+                    self.events[-1].text += char
+                else:
+                    # create a new TextEvent if there are no events yet
+                    # or if there is a new line
+                    self.events.append(TextEvent(char, x, y, w, h))
+                return
+            # char was found in a loaded JSON font => new behaviour
+            if self.events:
+                x_diff = x - (self.events[-1].x + self.events[-1].w)
+                if x_diff < 0:
+                    x_diff = -x_diff
```

Comment:
In which situation can this happen?
What would be the resulting textevent?

Reply:
I can see at least two cases where **x_diff** can be negative:

1. when this char is the first letter of next word on next line: Y will be increased (usually by font height) but X value will be also modified (usually to X starting position of next line).
2. when we display some letters (like **'y'** or **'j'**, which in some fonts can have a negative **x_offset** => in that case, current **x** can be a little bit < to previous **x+w**

In first case, the character need to be store on a new event, while in the second case it have to be concatenated to the current one.